### DISCUSSION TITLE
How can I see the Perl code generated code from a .ep template?

### DISCUSSION THREAD

**Discussion by janlimpens** [mojolicious/mojo#2005] [Q&A] [answered]
I am getting warnings in my .ep template that do not originate in my code, but seems to come from what mojo does with it.

Is there a way for me to see the thing mojo turns the .ep into?

*warning is a repeated 
`Warning: Use of "scalar" without parentheses is ambiguous`
So, I assume this is happening inside a loop

**Comment by kraih** [comment, marked as answer]
`MOJO_TEMPLATE_DEBUG=1`

**Reply by janlimpens** [reply]
Dankeschön!

The 'solution' to my problem was to put another 
`no warnings qw(abiguous);` at the top of my template. Don't know why the one inserted by the template wasn't picked up.

The use of `scalar` in the templates, as in
`$_O .= _escape scalar +  $my_variable ;`
puzzles me.

What trickery is happening there?
